tests/results/test_result_factory.py

```diff
@@ -82,6 +83,26 @@ def foo():
     assert result_factory.persist_result is True
 
 
+def test_root_flow_default_remote_storage():
+    @flow
+    async def foo():
+        result_fac = get_run_context().result_factory
+        return await result_fac.storage_block
+
+    block = S3(bucket_path="my-bucket")
+    block.save("my-result-storage")
+
+    with temporary_settings(
+        {
+            PREFECT_RESULTS_PERSIST_BY_DEFAULT: True,
+            PREFECT_DEFAULT_RESULT_STORAGE_BLOCK: block.get_block_type_slug(),
```

Comment:
as discussed async, this won't quite work, as the block type slug (e.g. `s3`) is only half of whats needed to load a block via `Block.load`

instead, the value of `PREFECT_DEFAULT_RESULT_STORAGE_BLOCK` would need to be something like

`s3/my-result-storage`

also, looking more at this I think we should either use a `LocalFileSystem` instead of `S3` (apologies for the misdirection there) or a dummy class that inherits from `WritableFileSystem`

that way we wouldnt have to worry about mocking network calls via boto etc

Reply:
I've implemented a `LocalFileSystem` within the unit test.